### PR TITLE
Add deprecation message for storage and compute optimized constraints

### PR DESCRIPTION
**Pull request by linkas45**

### CODE REVIEW

File: castai/resource_node_template.go [hunk 1]
```diff
@@ -223,6 +223,7 @@ func resourceNodeTemplate() *schema.Resource {
 							Optional:    true,
 							Default:     false,
 							Description: "Storage optimized instance constraint - will only pick storage optimized nodes if true",
+							Deprecated:  "Storage optimized constraint will be removed in the next major version and other resource for this functionality will have to be used.",
```
Comment (mikenorgate) [inline castai/resource_node_template.go:226]: This wording doesn't sound right, I feel it gives the impression we are going to be removing the constraint completely. Also we wont be replacing this with a different resource, just a different argument

Comment (linkas45) [reply castai/resource_node_template.go:226]: Hmm, agree, what do you think about this:
`Storage optimized constraint is deprecated and will be removed in the next major version. Other argument for the same functionality will be provided with the major version.`
Maybe you can think of other variations?

Comment (mikenorgate) [reply castai/resource_node_template.go:226]: `Storage optimized constraint is deprecated and will be replaced by a new argument in the next major release`?

Comment (linkas45) [reply castai/resource_node_template.go]: updated

**Review by furkhat — COMMENTED** [review]
Could new fields be added along with deprecation?

Then deprecation message could state "Deprecated.. use xyz instead". And the next major release would just delete deprecated fields.

What do you think?

**Comment by linkas45** [comment]
> Could new fields be added along with deprecation?
> 
> Then deprecation message could state "Deprecated.. use xyz instead". And the next major release would just delete deprecated fields.
> 
> What do you think?

Having 2 different properties changing single property will effect in terraform always seeing a difference when planning. If enum=true and boolean is not set (defaults to null). But then enum (because it's true) sets boolean value to true and then with another terraform plan there is a diff (true->null).

**Review by mikenorgate — APPROVED** [review]

**Pull request merged**